src/test/kotlin/org/rust/lang/core/type/RsGenericExpressionTypeInferenceTest.kt

```diff
@@ -151,6 +151,45 @@ class RsGenericExpressionTypeInferenceTest : RsTypificationTestBase() {
         }
     """)
 
+    fun testGenericFunction() = testExpr("""
+        fn f<T>(t: T) -> T {
+            return t;
+        }
```

Comment:
I usually prefer to keep test fixtures maximally compact, so this I would write as `fn f<T>(t: T) -> T { t }`.

Not 100% sure that this is a good idea, but I like the increased density in this case.